modules/EvseManager/EvseManager.hpp

```diff
@@ -196,13 +196,34 @@ class EvseManager : public Everest::ModuleBase {
         return powersupply_capabilities;
     }
 
-    void update_powersupply_capabilities(types::power_supply_DC::UpdateCapabilities caps) {
+    void update_powersupply_capabilities(types::power_supply_DC::Capabilities caps) {
         std::scoped_lock lock(powersupply_capabilities_mutex);
-        powersupply_capabilities.max_export_current_A = caps.max_export_current_A;
-        powersupply_capabilities.max_export_power_W = caps.max_export_power_W;
-
-        powersupply_capabilities.max_import_current_A = caps.max_import_current_A;
-        powersupply_capabilities.max_import_power_W = caps.max_import_power_W;
+        powersupply_capabilities = caps;
+
+        // Inform HLC layer about update of physical values
+        types::iso15118_charger::SetupPhysicalValues setup_physical_values;
+        setup_physical_values.dc_current_regulation_tolerance = powersupply_capabilities.current_regulation_tolerance_A;
+        setup_physical_values.dc_peak_current_ripple = powersupply_capabilities.peak_current_ripple_A;
+        setup_physical_values.dc_energy_to_be_delivered = 10000;
```

Comment:
would put this also into a helper function, so the length of the main function `update_powersupply_capabilities` gets shorter and the logic more clear.

Reply:
I think update_powersupply_capabilities() is not so long and splitting it up in 3-4 for functions would not really help with readability